here-naksha-lib-handlers/src/main/java/com/here/naksha/lib/handlers/val/MockValidationHandler.java

```diff
@@ -0,0 +1,149 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+package com.here.naksha.lib.handlers.val;
+
+import static com.here.naksha.lib.handlers.util.MockUtil.*;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.here.naksha.lib.core.IEvent;
+import com.here.naksha.lib.core.INaksha;
+import com.here.naksha.lib.core.NakshaContext;
+import com.here.naksha.lib.core.exceptions.XyzErrorException;
+import com.here.naksha.lib.core.models.XyzError;
+import com.here.naksha.lib.core.models.geojson.implementation.XyzFeature;
+import com.here.naksha.lib.core.models.geojson.implementation.XyzReference;
+import com.here.naksha.lib.core.models.naksha.EventHandler;
+import com.here.naksha.lib.core.models.naksha.EventHandlerProperties;
+import com.here.naksha.lib.core.models.naksha.EventTarget;
+import com.here.naksha.lib.core.models.storage.*;
+import com.here.naksha.lib.core.util.json.JsonSerializable;
+import com.here.naksha.lib.handlers.AbstractEventHandler;
+import com.here.naksha.lib.handlers.util.HandlerUtil;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MockValidationHandler extends AbstractEventHandler {
+
+  private static final Logger logger = LoggerFactory.getLogger(MockValidationHandler.class);
+  protected @NotNull EventHandler eventHandler;
+  protected @NotNull EventTarget<?> eventTarget;
+  protected @NotNull EventHandlerProperties properties;
+
+  private static final String MOCK_VIOLATIONS_FILE = "mock_data/dry_run_violations.json";
+  private static final TypeReference<List<XyzFeature>> LIST_FEATURE_TYPE_REF = new TypeReference<>() {};
+  private static final List<XyzFeature> mockViolations = parseJsonFile(MOCK_VIOLATIONS_FILE, LIST_FEATURE_TYPE_REF);
+  private static final int totalViolations = mockViolations.size();
+
+  public MockValidationHandler(
+      final @NotNull EventHandler eventHandler,
+      final @NotNull INaksha hub,
+      final @NotNull EventTarget<?> eventTarget) {
+    super(hub);
+    this.eventHandler = eventHandler;
+    this.eventTarget = eventTarget;
+    this.properties = JsonSerializable.convert(eventHandler.getProperties(), EventHandlerProperties.class);
+  }
+
+  /**
+   * The method invoked by the event-pipeline to process custom Storage specific read/write operations
+   *
+   * @param event the event to process.
+   * @return the result.
+   */
+  @Override
+  public @NotNull Result processEvent(@NotNull IEvent event) {
+    final NakshaContext ctx = NakshaContext.currentContext();
+    final Request<?> request = event.getRequest();
+
+    logger.info("Handler received request {}", request.getClass().getSimpleName());
+
+    return null;
```

Comment:
I know we're in Mock but maybe there is some sensible `Result` instance (like some empty one) that could be returned here?
Method signature promises that we won't return `null` here

Reply:
This method is not used at all, as handlers are not getting called as part of pipeline. Will return specific error for now.